packages/components/plugins/esbuild-css-module.ts

```diff
@@ -0,0 +1,63 @@
+/* eslint-disable import/no-extraneous-dependencies */
+import fs from "fs"
+import path from "path"
+import postcss from "postcss"
+import postcssModules from "postcss-modules"
+
+export default {
+  name: "css-module",
+  setup(build): void {
+    build.onResolve({ filter: /\.module\.css$/, namespace: "file" }, args => ({
+      path: `${args.path}#css-module`,
+      namespace: "css-module",
+      pluginData: {
+        pathDir: path.join(args.resolveDir, args.path),
+      },
+    }))
+    build.onLoad(
+      { filter: /#css-module$/, namespace: "css-module" },
+      async args => {
+        const { pluginData } = args as {
+          pluginData: { pathDir: string }
+        }
```

Comment:
Does this work? To avoid typecasting. Although of course, I don't know if `args` should be something else (ideally we'd have an actual type from somewhere).

```suggestion
      async ({ pluginData }: { pluginData: { pathDir: string } }) => {
```

Reply:
Can try, it's a copy pasta 😛